src/eth_provider/pending_pool.rs

```diff
@@ -0,0 +1,18 @@
+use crate::eth_provider::provider::EthDataProvider;
+use eyre::Result;
+use tokio::time::{sleep, Duration};
+
+pub async fn start_retry_service<SP>(eth_provider: EthDataProvider<SP>) -> Result<()>
+where
+    SP: starknet::providers::Provider + Send + Sync,
+{
+    // Start an infinite loop.
+    loop {
+        // Call the retry_transactions method
+        if let Err(err) = eth_provider.retry_transactions().await {
+            return Err(eyre::eyre!("Error while retrying transactions: {:?}", err));
```

Comment:
as discussed, let's remove this and just look it in order to avoid killing the service

Reply:
Just fixed that, I've used two rules here:
- If we encounter an explicit conversion error during the conversion in the retry loop, then we delete the transaction from the pending pool because we know that it is invalid.
- The other errors cannot be judged as drastically (potentially database connection error, provider error, etc.) so we just log them in the retry loop and start again.